### PR TITLE
Create Ukrainian 380.txt

### PR DESCRIPTION
**Pull request by burunduk**
Ukrainian geocoding data

### CODE REVIEW

**Comment by burunduk** [comment]
Sorted.
Name of region get from [Ministry of Foreign Affairs of Ukraine](http://mfa.gov.ua/en/about-ukraine/info/regions)
Also, this information can be obtained from the: [wiki](https://en.wikipedia.org/wiki/Oblasts_of_Ukraine)

**Comment by RamGopalBishnoi** [comment]
sir, Good evening. there is no mention about  subject why it went wrong  i 'm with you this pul may mute or delete from my device
Sent from my Samsung Galaxy smartphone.
-------- Original message --------From: kkeshava <notifications@github.com> Date: 11/05/2018  10:27 a.m.  (GMT+05:30) To: googlei18n/libphonenumber <libphonenumber@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: Re: [googlei18n/libphonenumber] Create Ukrainian 380.txt (#2195) 
@kkeshava requested changes on this pull request.



In resources/geocoding/uk/380.txt:
> +38045|Київська обл.
+38046|Чернігівська обл.
+38047|Черкаська обл.
+38048|Одеська обл.
+38051|Миколаївська обл.
+38052|Кіровоградська обл.
+38053|Полтавська обл.
+38054|Сумська обл.
+38055|Херсонська обл.
+38056|Дніпропетровська обл.
+38057|Харківська обл.
+38061|Запорізька обл.
+38062|Донецька обл.
+38064|Луганська обл.
+38065|АР Крим
+38069|Севастопіль

ukrtelecom webiste is more granular with city and region details. However this site doesn't give details of few regions that are mentioned in ITU doc eg. prefix 31|Zakarpattia, 44|Kyiv, 65|Crimea and 69|Sevastopol. So how did you get the Ukrainian names for these?
We need the geocoding data in ascending order. Currently it is mixed up. Please correct it.

File: resources/geocoding/uk/380.txt [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+3803735|Сторожинець, Чернівецька обл.
+3803736|Кіцмань, Чернівецька обл.
+3803737|Заставна, Чернівецька обл.
+3803738|Путила, Чернівецька обл.
+3803739|Сокиряни, Чернівецька обл.
+3803740|Герца, Чернівецька обл.
+3803741|Новодністровськ, Чернівецька обл.
+38038|Хмельницька обл.
+380382|Хмельницький, Хмельницька обл.
+3803822|Хмельницький, Хмельницька обл.
```
Comment (kkeshava) [inline resources/geocoding/uk/380.txt]: Remove 3822

File: resources/geocoding/uk/380.txt [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+3804579|Славутич, Київська обл.
+3804591|Іванків, Київська обл.
+3804594|Бровари\Калита, Київська обл.
+3804595|Бориспіль, Київська обл.
+3804596|Вишгород\Димер\Пірнове, Київська обл.
+3804597|Буча\Ворзель\Гостомель\Ірпінь\Коцюбинське, Київська обл.
+3804598|Боярка\Вишневе, Київська обл.
+38046|Чернігівська обл.
+380462|Чернігів, Чернігівська обл.
+3804622|Чернігів, Чернігівська обл.
```
Comment (kkeshava) [inline resources/geocoding/uk/380.txt]: Remove 4622

**Comment by burunduk** [comment]
Ready.
Can we use prefix "м." for the city?

**Comment by kkeshava** [comment]
What does it mean actually? Also for 65 you have mentioned as "AP", what does it mean?
Generally we do not add any such prefix for cities/regions.

**Comment by burunduk** [comment]
"АР" abr. "Автономна республіка" [(Autonomous Republic of Crimea)](https://en.wikipedia.org/wiki/Crimea)
"м." abr. "місто" - (eng)City.

File: resources/geocoding/uk/380.txt [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+3806147|Куйбишеве, Запорізька
+3806153|Бердянськ, Запорізька
+3806156|Велика Білозерка, Запорізька
+3806162|Розівка, Запорізька
+3806165|Пологи, Запорізька
+3806175|Василівка/Дніпрорудне, Запорізька
+3806178|Токмак, Запорізька
+380619|Мелітополь, Запорізька
+3806192|Мелітополь, Запорізька
+38062|Донецьк/Донецька
```
Comment (kkeshava) [inline resources/geocoding/uk/380.txt]: As per online search it seems like Донецьк is more commonly used than Донецька. So please change it everywhere. Also keep single entry for 62 i.e 62|Донецьк

File: resources/geocoding/uk/380.txt [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+3806273|Вугледар, Донецька
+3806274|Артемівськ/Сіверськ, Донецька
+3806275|Докучаєвськ, Донецька
+3806277|Добропілля, Донецька
+3806278|Марʼїнка, Донецька
+3806279|Тельманове, Донецька
+380629|Маріуполь, Донецька
+3806296|Новоазовськ, Донецька
+3806297|Мангуш/Ялта, Донецька
+38064|Луганська
```
Comment (kkeshava) [inline resources/geocoding/uk/380.txt]: As per Wiki and online search it seems like Луганськ is more commonly used than Луганська. So please change it everywhere.

File: resources/geocoding/uk/380.txt [hunk 5]
```diff
@@ -0,0 +1,584 @@
...
+3806462|Білокуракине, Луганська
+3806463|Новопсков, Луганська
+3806464|Марківка, Луганська
+3806465|Мілове, Луганська
+3806466|Біловодськ, Луганська
+3806471|Сватове, Луганська
+3806472|Станиця Луганська, Луганська
+3806473|Словʼяносербськ, Луганська
+3806474|Попасна, Луганська
+38065|АР Крим
```
Comment (kkeshava) [inline resources/geocoding/uk/380.txt]: As ITU and Wiki states as Crimea and no need to mention AP here.

File: resources/geocoding/uk/380.txt [hunk 6]
```diff
@@ -0,0 +1,584 @@
...
+3805158|Баштанка, Миколаївська
+3805159|Єланець, Миколаївська
+3805161|Первомайськ, Миколаївська
+3805162|Снігурівка, Миколаївська
+3805163|Веселинове, Миколаївська
+3805164|Казанка, Миколаївська
+3805167|Нова Одеса, Миколаївська
+3805168|Березнегувате, Миколаївська
+38052|Кіровоградська
+380522|Кіровоград, Кіровоградська
```
[the remark this answers is not in the document]
Comment (burunduk) [reply resources/geocoding/uk/380.txt]: Area is stil name Kirovohrad region.
May be:
52|Kirovohrad (region name)
522|Kropyvnytskyi (city name)?

**Comment by kkeshava** [comment]
Hi,
We haven't received a response from you yet. Would it be possible for you to provide an update in order for us to move forward with fixing this issue?

**Comment by burunduk** [comment]
I have a question. Are the region and the city the same code I have to write? What is the name of a region or city?
For example: 38037 is a city of Chernivtsi, but this is the Chernivtsi region.

**Comment by kkeshava** [comment]
As per ITU 37 is assigned to Chernivtsi. So we generally keep that and other specific prefixes under 37. Here 380372 and 3803722 is assigned to Chernivtsi(city) so instead of writing Chernivtsi(city), Chernivtsi(region) we optimise it so that it falls back to 37|Chernivtsi

Please let us know once you have done the other corrections. I am still able to find already addressed issues back https://github.com/googlei18n/libphonenumber/pull/2195/files
Eg. 
38031|Закарпаття should be 38031|Закарпатська
38032|Львів should be Львів/Львівська

These were correct in older commit 86b4ef2. Could you please check once.

**Comment by burunduk** [comment]
Ready.

**Comment by kkeshava** [comment]
Hi,

The new commit also does not have all the changes that are suggested. To avoid the review iterations further, we will update the Ukraine geocode mappings as part of next release which is planned for next week. Can we add your name to the [CONTRIBUTORS](https://github.com/googlei18n/libphonenumber/blob/master/CONTRIBUTORS) as you have provided the data?

**Comment by burunduk** [comment]
Yes please.
Regarding the changes: it will be easier to enter fewer data at a time.

**Comment by kkeshava** [comment]
I agree also it will require lot of back and forth to review :)
 
Thank you for contributing to libphonenumber! 
As agreed we will be closing the PR and the fix for this will be available in our next upcoming release.

**Comment by RamGopalBishnoi** [comment]
Sent from my Samsung Galaxy smartphone.

**Pull request closed without merging**